libraries/standard/mqtt/src/mqtt.c

```diff
@@ -865,9 +874,12 @@ static MQTTStatus_t handlePublishAcks( MQTTContext_t * pContext,
 
     if( status == MQTTSuccess )
     {
+        pDeserialized->packetIdentifier = packetIdentifier;
```

Comment:
`pDeserialized->pPublishInfo = NULL`?

Reply:
`handleIncomingAck` already sets `deserializedInfo.pPublishInfo = NULL;` before this function is called, so I didn't think it necessary to set it again.

Reply:
Yes, I think it should be set as this function shouldn't be aware of the implementation details of its calling function. More importantly, this function shouldn't depend on the calling function for setting the parameter correctly.

Reply:
Handled when pointer was removed